RISC-V: ACPI: Enable SPCR table for console output on RISC-V

uestc-gr opened the pull request:
SPCR表提供了一个标准化的方式来描述系统中可用的串行端口，并指定了用于控制台输出的串行端口。
这使得操作系统能够在启动时自动检测并使用正确的串行端口进行控制台输出，而无需用户手动指定“
console=xxx”等参数。
RISC-V架构在许多方面与ARM64和X86有相似之处，特别是在系统启动和硬件初始化阶段，
因此同样的SPCR代码可以被复用于RISC-V架构。这意味着开发者可以利用已有的代码基础，减少开发工作量，
同时也能快速实现RISC-V系统在启动时的控制台输出功能

fixed: <https://github.com/RVCK-Project/rvck/issues/55>

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/61/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/64/consoleFull

uestc-gr committed:
RISC-V: ACPI: Enable SPCR table for console output on RISC-V

mainline inclusion
from mainline-v6.11-rc1
commit 38738947db38520b58b7dae64bd0eec513e83139
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/74

--------------------------------

The ACPI SPCR code has been used to enable console output for ARM64 and
X86. The same code can be reused for RISC-V. Furthermore, SPCR table is
mandated for headless system as outlined in the RISC-V BRS
Specification, chapter 6.

Signed-off-by: Sia Jee Heng <jeeheng.sia@starfivetech.com>
Reviewed-by: Sunil V L <sunilvl@ventanamicro.com>
Link: https://lore.kernel.org/r/20240502073751.102093-2-jeeheng.sia@starfivetech.com
Signed-off-by: Palmer Dabbelt <palmer@rivosinc.com>
Signed-off-by: Gao Rui <gao.rui@zte.com.cn>

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/62/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/65/consoleFull

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/225

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/226
